core/pom.xml

```diff
@@ -76,12 +76,31 @@
         <artifactId>maven-javadoc-plugin</artifactId>
         <configuration>
           <failOnError>true</failOnError>
-          <failOnWarnings>true</failOnWarnings>
         </configuration>
       </plugin>
     </plugins>
   </build>
 
+  <profiles>
+    <profile>
+      <id>javadoc-warning-fail-java8</id>
+      <activation>
+        <jdk>8</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <configuration>
+              <failOnWarnings>true</failOnWarnings>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
```

Comment:
There should be no need to do this here and further down the PR, it should be inherited appropriately.

Reply:
Done